src/generators/dom/index.js

```diff
@@ -168,7 +169,7 @@ export default function dom ( parsed, source, options ) {
 	if ( templateProperties.oncreate ) {
 		builders.init.addBlock( deindent`
 			if ( options._root ) {
-				options._root._renderHooks.push({ fn: ${generator.alias( 'template' )}.oncreate, context: this });
+				options._root._renderHooks.push( ${generator.alias( 'template' )}.oncreate.bind( this ) );
```

Comment:
Since `.bind` is slower than pretty much every other option of calling a function later, we could do something like

```javascript
var self = this;
options._root._renderHooks.push( function () {
  ${ generator.alias( ' template' ) }.oncreate.call ( self );
} );
```

which would be slightly faster. Here's a test showing this: https://jsperf.com/bind-vs-self-closure (interestingly enough the calling of a `.bind`ed function is no slower than the closure in firefox, but in chrome it's 10x slower)